tests/test_processors.py

```diff
@@ -321,3 +331,83 @@ def aggregateRating(self):
     assert page.aggregateRating == AggregateRating(
         ratingValue=3.8, bestRating=10, reviewCount=5
     )
+
+
+@pytest.mark.parametrize(
+    "input_value,expected_value",
+    [
+        (None, None),
+        ([], []),
+        ("https://www.url.com/img.jpg", [Image(url="https://www.url.com/img.jpg")]),
+        (
+            [
+                Image("https://www.url.com/img1.jpg"),
+                Image("https://www.url.com/img2.jpg"),
+            ],
+            [
+                Image("https://www.url.com/img1.jpg"),
+                Image("https://www.url.com/img2.jpg"),
+            ],
+        ),
+        (
+            ["https://www.url.com/img1.jpg", "https://www.url.com/img2.jpg"],
+            [
+                Image("https://www.url.com/img1.jpg"),
+                Image("https://www.url.com/img2.jpg"),
+            ],
+        ),
+        (
+            [
+                {"url": "https://www.url.com/img1.jpg"},
+                {"url": "https://www.url.com/img2.jpg"},
+            ],
+            [
+                Image("https://www.url.com/img1.jpg"),
+                Image("https://www.url.com/img2.jpg"),
+            ],
+        ),
+    ],
+)
+def test_images(input_value, expected_value):
+    class ImagesPage(BasePage):
+        @field(out=[images_processor])
+        def images(self):
+            return input_value
+
+    page = ImagesPage(base_url)  # type: ignore[arg-type]
+    assert page.images == expected_value
+
+
+def test_images_page():
+    class MyProductPage(ProductPage):
+        @field
+        def images(self):
+            return self.css("img::attr(href)").getall()
+
+    response = HttpResponse(
+        url="http://www.example.com/",
+        body="<html><body><img href='https://www.url.com/img.jpg'></body></html>".encode(),
+    )
+    page = MyProductPage(response=response)
+    assert page.images == [Image(url="https://www.url.com/img.jpg")]
+
+
+@pytest.mark.parametrize(
+    "input_value,expected_value",
+    [
+        (100, "100.00"),
+        (None, None),
+        ([], []),
+        ({}, {}),
+        (22.9, "22.90"),
+        (22.0, "22.00"),
```

Comment:
Should we add a test scenario that confirms that strings are left as is, e.g. `("22.9", "22.9")`?

Reply:
Added new test cases in 12cb8efadd59d399b41bf59de6b0d721d4d10efb